cpp/src/arrow/csv/chunker_test.cc

```diff
@@ -261,5 +291,43 @@ TEST_P(BaseChunkerTest, EscapingNewline) {
   }
 }
 
+TEST_P(BaseChunkerTest, ParseSkip) {
+  {
+    auto csv = MakeCSVData({"ab,c,\n", "def,,gh\n", ",ij,kl\n"});
+    ASSERT_NO_FATAL_FAILURE(AssertSkip(csv, 1, 0, 15));
```

Comment:
I don't think `ASSERT_NO_FATAL_FAILURE` is useful, but we can keep it if you want.

Reply:
Without ASSERT_NO_FATAL_FAILURE the test will continue to execute even if AssertSkip fails.

With gtest assert failures in called functions do not cause the test to exit unless it is wrapped by ASSERT_NO_FATAL_FAILURE

Reply:
Ok, but how is that a problem? It would just accumulate failures instead of stopping at the first one.
Regardless, this is not important :-)

Reply:
It can be a problem for some tests which might cause a segfault or have bad behavior if test execution fails after an ASSERT fails, I know this is not one of those types of tests. Normally ASSERT failures halt test execution for non halting checks it is standard to use the `EXPECT_` macros